packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -159,12 +159,17 @@ const SceneView = ({
 }: SceneViewProps) => {
   const { route, navigation, options, render } = descriptor;
 
-  let {
-    animation,
-    animationMatchesGesture,
-    fullScreenGestureEnabled,
-    presentation = 'card',
-  } = options;
+  let { animation, animationMatchesGesture, fullScreenGestureEnabled } =
+    options;
+
+  if (
+    !options.presentation &&
+    previousDescriptor?.options.presentation === 'modal'
+  ) {
+    options.presentation = 'modal';
```

Comment:
We shouldn't mutate props/options.

Reply:
good point, thanks
I've changed to another approach with an additional flag